File: fixtures/sample.js
// Fixture: exercise javascript highlighting + symbol picker.

const MAX_RETRIES = 3;
const GREETING = "hello, world";

class Person {
    constructor(name, age) {
        this.name = name;
        this.age = age;
    }

    greet() {
        return `hi, ${this.name}! you are ${this.age}`;
    }
}

function double(x) {
    return x * 2;
}

const shout = (s) => `${s.toUpperCase()}!!!`;

async function fetchUser(id) {
    const res = await fetch(`/api/users/${id}`);
    if (!res.ok) throw new Error(`failed: ${res.status}`);
    return res.json();
}

function main() {
    const p = new Person("Ada", 36);
    const counts = new Map([["hits", 0]]);
    for (let i = 0; i < MAX_RETRIES; i++) {
        if (i % 2 === 0) {
            console.log(GREETING, "->", p.greet());
        } else {
            console.error(shout(GREETING));
        }
    }
    return double(21);
}

main();

File: fixtures/sample.rs
//! Fixture: exercise rust highlighting + symbol picker.

use std::collections::HashMap;
use std::fmt;

const MAX_RETRIES: u32 = 3;
static GREETING: &str = "hello, world";

pub trait Greet {
    fn greet(&self) -> String;
}

#[derive(Debug, Clone)]
pub struct Person {
    pub name: String,
    pub age: u32,
}

impl Greet for Person {
    fn greet(&self) -> String {
        format!("hi, {}! you are {}", self.name, self.age)
    }
}

pub enum Status {
    Online,
    Offline,
    Away(String),
}

macro_rules! shout {
    ($s:expr) => { format!("{}!!!", $s.to_uppercase()) };
}

mod helpers {
    pub fn double(x: i64) -> i64 { x * 2 }
}

fn main() {
    let p = Person { name: "Ada".into(), age: 36 };
    let mut counts: HashMap<&str, u32> = HashMap::new();
    counts.insert("hits", 0);
    for i in 0..MAX_RETRIES {
        if i % 2 == 0 {
            println!("{} -> {}", GREETING, p.greet());
        } else {
            eprintln!("{}", shout!(GREETING));
        }
    }
    let _ = helpers::double(21);
}
